BookingV2.Logic/Services/RealEstateService.cs

```diff
@@ -0,0 +1,167 @@
+using BookingV2.Logic.Contract;
+using BookingV2.Logic.Responses;
+using BookinV2.Data;
+using BookinV2.Data.Entities.RealEstateEntities;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace BookingV2.Logic.Services
+{
+    public class RealEstateService : IRealEstateService
+    {
+        private readonly BookingV2DBContext _context;
+        private readonly ILogger<RealEstateService> _logger;
+
+        public RealEstateService(BookingV2DBContext context, ILogger<RealEstateService> logger)
+        {
+            this._context = context ?? throw new ArgumentNullException(nameof(context));
+            this._logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public async Task<ApiResponse<IEnumerable<RealEstate>>> GetAllAsync()
+        {
+            var response = new ApiResponse<IEnumerable<RealEstate>>();
+
+            try
+            {
+                if (this._context.RealEstates == null)
+                {
+                    response.AddError("RealEstate DbSet is null.");
+                    return response;
+                }
+
+                response.Response = await this._context.RealEstates.ToListAsync();
+            }
+            catch (Exception ex)
```

Comment:
I think that we dont need to use `try catch` block here as we dont have any obvious exception which we need to handle, so I suggest just to remove it